src/Workspaces/Core/Portable/Shared/Extensions/IParameterSymbolExtensions.cs

```diff
@@ -11,7 +11,14 @@ internal static class IParameterSymbolExtensions
     {
         public static bool IsRefOrOut(this IParameterSymbol symbol)
         {
-            return symbol.RefKind != RefKind.None;
+            switch(symbol.RefKind)
```

Comment:
seems like ```return  symbol.RefKind == Ref || symbol.RefKind == Out``` would be sufficient here.  Switch seems like overkill.

Reply:
in case of this symbol calculating refkind from child symbols, this saves a few cycles. Also, same as saving into a temp followed by an if statement.

---
In reply to: [141181968](https://github.com/dotnet/roslyn/pull/22285#discussion_r141181968) [](ancestors = 141181968)